src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -226,6 +226,7 @@ async def pex_from_targets(request: PexFromTargetsRequest, python_setup: PythonS
             if tgt.has_field(PythonRequirementsField)
         ),
         additional_requirements=request.additional_requirements,
+        apply_constraints=True,
```

Comment:
N.B.: This balancing addition of `apply_constraints=True` for the removal on line [312 on the LHS](https://github.com/pantsbuild/pants/pull/12883/files#diff-796c32e79f63384b20df5eaaf3add13cff74e21470a478675503a5bf10bf77c9L312) was not present in the original PR: https://github.com/pantsbuild/pants/pull/12675/commits/62320df547f07e9a90ff6400638c2950f0b9d032#diff-796c32e79f63384b20df5eaaf3add13cff74e21470a478675503a5bf10bf77c9

Reply:
... and so could really use a hard second look.

Reply:
Yeah, this looks right. Note that we ban constraints files if you use a lockfile (for now), so no weird interaction there.

Reply:
Ok, thanks for the second set of eyes.

Reply:
Ah, yeah, the default for `create_from_requirement_fields` is True, so this was already working correctly.